File: AndroidWheelViewExample/app/src/main/java/com/example/shakil/androidwheelviewexample/MainActivity.java
package com.example.shakil.androidwheelviewexample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Toast;

import java.util.Arrays;

import kotlin.Unit;
import kotlin.jvm.functions.Function1;
import me.sujanpoudel.wheelview.WheelView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final WheelView wheelView = findViewById(R.id.wheel_view);
        wheelView.setTitles(Arrays.asList("One", "Two", "Three", "Four", "Five", "Six"));
        wheelView.setSelectListener(new Function1<Integer, Unit>() {
            @Override
            public Unit invoke(Integer integer) {
                showToast(integer);
                return Unit.INSTANCE;
            }

            private void showToast(Integer index) {
                Toast.makeText(MainActivity.this, "" + wheelView.getTitles().get(index), Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: KotlinWheelViewExample/app/src/main/java/com/example/shakil/kotlinwheelviewexample/MainActivity.kt
package com.example.shakil.kotlinwheelviewexample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import me.sujanpoudel.wheelview.WheelView
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val wheelView = findViewById<WheelView>(R.id.wheel_view)
        wheelView.titles = Arrays.asList("One", "Two", "Three", "Four", "Five", "Six")

        wheelView.selectListener = object : Function1<Int?, Unit> {
            override fun invoke(p1: Int?) {
                if (p1 != null) {
                    showToast(p1)
                }
            }

            private fun showToast(index: Int) {
                Toast.makeText(this@MainActivity, "" + wheelView.titles[index], Toast.LENGTH_SHORT).show()
            }

        }
    }
}
